packages/remix-react/components.tsx

```diff
@@ -1248,16 +1248,29 @@ export interface SubmitFunction {
  *
  * @see https://remix.run/api/remix#usesubmit
  */
-export function useSubmit(): SubmitFunction {
-  return useSubmitImpl();
+export function useSubmit(options?: SubmitOptions): SubmitFunction {
+  let key = undefined; // no key for global submit
+  return useSubmitImpl(key, options);
 }
 
-let defaultMethod = "get";
-let defaultEncType = "application/x-www-form-urlencoded";
-
-export function useSubmitImpl(key?: string): SubmitFunction {
+const DEFAULT_METHOD = "get";
+const DEFAULT_ENC_TYPE = "application/x-www-form-urlencoded";
```

Comment:
Switched these to `const` and upper-case because otherwise typescript wasn't happy about default-assigning `submitMethod` to `DEFAULT_METHOD` below. Using `const` here allows typescript to narrow this down from a type of `string` to `"get"` which fits in the `FormMethod` type.